test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2524,11 +1717,6 @@ describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
       expect(fileSystemCfConfig.FileSystemConfigs).to.deep.equal([
         { Arn: arn, LocalMountPath: localMountPath },
       ]);
-      expect(iamRolePolicyStatements).to.deep.include({
```

Comment:
Why this check was removed?

Reply:
I've added the explanation in the PR body: "The check for IAM statements in EFS-related test was removed, because it's duplicated with similar test should support functions[].fileSystemConfig (with vpc configured on function) and following the convetion, it seems like all adjustments to the default IAM role should be tested as a part of Function properties test suite.". In addition, it requires a setup of a different fixture, as the current one has `role` set on provider level, which will result in no extra IAM statements to be added to the role. 

What is your general opinion on testing Role statements in provider-related tests? If we'd like to do it, I believe a few other tests should also be expanded and include such checks.

Reply:
Sorry, I didn't read PR note (I should have!)

Anyway I believe we should confirm on that in those two tests, to be sure that configuration as put in both locations ensures VPC config (note that both cases are triggered by different `runServerless` run)

Reply:
Thanks, I've proposed splitting provider-related tests into two sections, depending on if they use default IAM role or not and I've brought back the assertions for EFS config. In addition, I also added checks for statements in tracing and kms tests. Let me know what do you think about this approach, thanks in advance :bow: